api/src/org/labkey/api/exp/property/DomainUtil.java

```diff
@@ -917,6 +935,32 @@ public static ValidationException updateDomainDescriptor(GWTDomain<? extends GWT
         return validationException;
     }
 
+    public static @Nullable String validateDomainName(@NotNull String domainName, String kindName, boolean supportNamingPattern)
+    {
+        String prefix = "Invalid " + kindName + " name. ";
+
+        if (StringUtils.isBlank(domainName))
+            return "Domain name must not be blank";
+
+        char start = domainName.charAt(0);
+        if (!Character.isLetterOrDigit(start))
+            return prefix + "Domain name must start with a letter or a number character.";
```

Comment:
```suggestion
            return prefix + "Domain name must start with a letter or a number.";
```
(or it could be "must not start with a letter or a digit")